accepted/2020/form-factors.md

```diff
@@ -0,0 +1,206 @@
+
+* [Platform Compatibility Analyzer](https://github.com/dotnet/platform-compat) -- identifies uses of APIs that are problematic on specific platforms
+* [.NET Portability Analyzer](https://docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer) -- identifies APIs missing on specific target .NET platforms
+
+The analyzers will enable us to transition the current trial and error experience into a more predictable one. Once an application or a library builds successfully, without warnings, it will work during runtime. If there are no warnings and it does not work on a specific form factor due to unsupported functionality, we will treat it as a bug.
+
+The analyzers will be complemented by guidance and product features that allow mitigation of the identified issues. We will work within the .NET team and across the .NET ecosystem to implement the mitigations to make specific vertical scenarios (e.g. client-side Blazor) work well. The larger ecosystem effort is going to be a journey spanning multiple release cycles. We do not expect the entire .NET ecosystem to implement mitigations for specific form factors, either because the scenario is not relevant for the specific package or because the amount of work and redesign required.
+
+We are intentionally not trying to address the subsets of incompatible APIs by introducing new smaller TFMs that exclude these APIs. Building systems with walls leads to ecosystem fragmentation. We have done that in the past and we were not happy with the result, like for example with .NET Framework 4.0 and the "client profile" concept. Our focus is going to be on exposing a single set of broadly available and useful APIs defined by the [`net5` TFM](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md) introduced with .NET 5.0 and future versions of the platform that expand it.
+
+### IL Linking
+
+The [IL linker](https://github.com/mono/linker) can generate optimized applications that fail at runtime, due to not being able to reliably analyze various problematic code patterns (largely centered on reflection use). It is not possible or desirable to simply invest more to improve the linker, as a strategy. Instead, build-time source generators will be the recommended mitigation for arbitrary reflection use.
+
+The barrier for entry for building source generators is very high today. There is no standard solution and only a few libraries can afford to build a custom one. Razor pages, gRPC, WPF data binding, and Data-Contract serializers each have their own custom build-time source generator solutions today. First-class Roslyn support for [source generators](https://github.com/dotnet/roslyn/blob/master/docs/features/source-generators.md) will lower the barrier of entry for components that need to generate source code.
```

Comment:
> Razor pages, gRPC, WPF data binding, and Data-Contract serializers each have their own custom build-time source generator solutions today.

I feel like this should be UWP data binding, as WPF doesn't support compile time binding (x:Bind)

Reply:
You are right that WPF does not use build time source generation for data bindings. WPF uses the build time source generation for other reason.